larpandoracontent/LArHelpers/LArFormattingHelper.cc

```diff
@@ -0,0 +1,184 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArFormattingHelper.cc
+ *
+ *  @brief  Implementation of the formatting helper class.
+ *
+ *  $Log: $
+ */
+
+#include <iomanip>
+
+#include "Pandora/StatusCodes.h"
+
+#include "larpandoracontent/LArHelpers/LArFormattingHelper.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+    void LArFormattingHelper::SetStyle(const LArFormattingHelper::Style &style, std::ostream &stream)
+    {
+        LArFormattingHelper::PrintFormatCharacter(static_cast<unsigned int>(style), stream);
+    }
+   
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::SetColour(const LArFormattingHelper::Colour &colour, std::ostream &stream)
+    {
+        LArFormattingHelper::PrintFormatCharacter(static_cast<unsigned int>(colour), stream);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::ResetStyle(std::ostream &stream)
+    {
+        LArFormattingHelper::SetStyle(REGULAR, stream);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::ResetColour(std::ostream &stream)
+    {
+        LArFormattingHelper::SetColour(DEFAULT, stream);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::Reset(std::ostream &stream)
+    {
+        LArFormattingHelper::ResetColour(stream);
+        LArFormattingHelper::ResetStyle(stream);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintFormatCharacter(const unsigned int &code, std::ostream &stream)
+    {
+        stream << LArFormattingHelper::GetFormatCharacter(code);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    std::string LArFormattingHelper::GetFormatCharacter(const unsigned int &code)
+    {
+        return ("\x1B[" + std::to_string(code) + "m");
```

Comment:
Is there a way to avoid the literals, e.g. can they be accessed from a standard header instead?

Reply:
I did some googling, but I can't find any standard header. I don't know for sure though!

Reply:
With a quick bit of googling, I did spot the following:

>These are ANSI escape sequences (also known as VT100 codes) are an early standardisation of control codes pre-dating ASCII.
>
>The escape sequence \x1BE, or Esc+E, is NEL or "Next line", and is used on older terminals and mainframes to denote CR+LF, or \r\n.

```\r\n``` is definitely more recognisable, but still a hard-coded string literal!

Reply:
Could just go with something like the following example, from [msdn](https://msdn.microsoft.com/en-us/library/69ze775t.aspx)?

```
    char newline = '\n';  
    char tab = '\t';  
    char backspace = '\b';  
    char backslash = '\\';  
    char nullChar = '\0';  
  
    cout << "Newline character: " << newline << "ending" << endl; // Newline character:  
                                                                  //  ending  
    cout << "Tab character: " << tab << "ending" << endl; // Tab character : ending  
    cout << "Backspace character: " << backspace << "ending" << endl; // Backspace character : ending  
    cout << "Backslash character: " << backslash << "ending" << endl; // Backslash character : \ending  
    cout << "Null character: " << nullChar << "ending" << endl; //Null character:  ending  
```

Reply:
I think `\x1BE = ESC+E` (for next line as you say `\r\n`), but in this case we want `\x1B[ = ESC+[` which is the "control sequence introducer" for changing styles / colours.

I can make const static variables for the control characters, however, they are only used in one place. What do you think?